File: src/test/resources/generics/GenericTypeReference.cs
using System;
using System.Collections;
using System.Collections.Generic;
using nonamespace;

namespace generics {
    public class GenericTypeReference<T> {
        private static readonly Type myclass = typeof(GenericTypeReference<>);

        public static void test(object o) {
            if (o.GetType() == typeof(GenericTypeReference<>)) {
                System.Console.Out.WriteLine("1");
            }
            System.Console.Out.WriteLine(typeof(GenericTypeReference<>));
            System.Console.Out.WriteLine(typeof(IList));
            System.Console.Out.WriteLine(typeof(List<>));
            System.Console.Out.WriteLine(typeof(EnumBase));
        }
    }
}

File: src/test/resources/generics/GenericTypeReference.java
package generics;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;
import java.util.Set;

public class GenericTypeReference<T> {
    private static final Class myclass = GenericTypeReference.class;

    public static void test(Object o) {
        if (o.getClass() == GenericTypeReference.class) {
            System.out.println("1");
        }
        System.out.println(GenericTypeReference.class);
        System.out.println(List.class);
        System.out.println(Map.class);
        System.out.println(Set.class);
        System.out.println(ArrayList.class);
        System.out.println(Enum.class);
    }

}